server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDeciderTests.java

```diff
@@ -117,6 +110,23 @@ public void testWriteLoadDeciderDisabled() {
      * Test the {@link WriteLoadConstraintDecider#canAllocate} implementation.
      */
     public void testWriteLoadDeciderCanAllocate() {
+        internalDeciderAllocationCheck((decider, shardRouting, node, allocation) -> decider.canAllocate(shardRouting, node, allocation));
+    }
+
+    /**
+     * Test the {@link WriteLoadConstraintDecider#canForceAllocateDuringReplace} implementation.
+     */
+    public void testWriteLoadDeciderCanForceAllocateDuringReplace() {
+        internalDeciderAllocationCheck(
+            (decider, shardRouting, node, allocation) -> decider.canForceAllocateDuringReplace(shardRouting, node, allocation)
+        );
+    }
+
+    interface DeciderAllocation {
+        Decision canAllocate(WriteLoadConstraintDecider decider, ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation);
+    }
+
+    void internalDeciderAllocationCheck(DeciderAllocation decider) {
```

Comment:
I generally dislike prefixes like `internal*` because they are super uninformative: they add no more information than someone wanted to wrap/refactor the logic at some point (how I've seen it used in our codebase). Fluff words bad :) 

Perhaps simply `checkDeciderCanAllocateDecisions` -- and pls add a brief method comment describing that this lets us test both decider methods.

Reply:
Yeah I think I agree, perhaps `runCommonDeciderTests(DeciderInvocationStrategy strategy)` if using above naming?